server/src/internalClusterTest/java/org/elasticsearch/discovery/DiscoveryDisruptionIT.java

```diff
@@ -215,13 +215,7 @@ public void testJoinWaitsForClusterApplier() throws Exception {
         final var victimName = randomValueOtherThan(masterName, () -> randomFrom(internalCluster().getNodeNames()));
         logger.info("--> master [{}], victim [{}]", masterName, victimName);
 
-        // drop the victim from the cluster with a network disruption
-        final var masterTransportService = (MockTransportService) internalCluster().getInstance(TransportService.class, masterName);
-        masterTransportService.addFailToSendNoConnectRule(internalCluster().getInstance(TransportService.class, victimName));
```

Comment:
One question to understand this mock. Wouldn't this `addFailToSendNoConnectRule` prevent/drop reconnect attempts of the victim (to master)? How can the victim rejoin after this failure has been added to the transport service of master (and before it is cleared)?

Reply:
No, this only prevents the master from sending/connecting to the victim, the connections in the opposite direction are unaffected.

Reply:
... but also the victim was rejoining after this rule was cleared.